### ISSUE TITLE
Add ignoreProperties to declaration-block-no-duplicate-custom-properties

### ISSUE THREAD

**Issue by calinoracation** [stylelint/stylelint#6763]
### What is the problem you're trying to solve?

Hitting errors on the rule due to my pattern of using customizable effects in my classes that duplicate properties. I know they are duplicated and am fine with it.

```tsx
[view-transition-new='screen'] {
  ${RelativeMoveEffect.apply({ endX: 0, startX: '200px', duration: 500 })};
  ${FadeEffect.apply({ startOpacity: 0, endOpacity: 1, delay: 50, duration: 350 })};
  animation: ${FadeEffect.animation}, ${RelativeMoveEffect.animation};
}
```

Each of these "effects" apply defaults, say a default `--view-transition_mix-blend-mode`. They can each also be used individually. In my case I'm totally fine with the last one "winning" and want to allow this exception.

### What solution would you like to see?

Add an `ignoreProperties` syntax like in other rules to let me prefix my exceptions.

```
ignoreProperties: ['/--view-transition']
```

**Comment by ybiquitous** [comment]
@calinoracation Thanks for the proposal. Sounds good to me. 👍🏼 

I've labeled the issue as **ready to implement**. Please consider [contributing](https://stylelint.io/contributing) if you have time.

There are [steps on how to add a new option](https://stylelint.io/developer-guide/rules#add-an-option-to-a-rule) in the Developer guide.
